core/src/main/java/org/apache/calcite/sql2rel/StandardConvertletTable.java

```diff
@@ -2049,6 +2054,21 @@ private static class TimestampAddConvertlet implements SqlRexConvertlet {
     }
   }
 
+  /** Convertlet that handles the BigQuery {@code DATETIME_TRUNC} and
+   * {@code TIMESTAMP_TRUNC} functions. */
+  private static class TruncConvertlet implements SqlRexConvertlet {
+    @Override public RexNode convertCall(SqlRexContext cx, SqlCall call) {
+      final RexBuilder rexBuilder = cx.getRexBuilder();
+      RexNode op1 = cx.convertExpression(call.operand(0));
+      RexNode op2 = cx.convertExpression(call.operand(1));
+      if (op1.getType().getSqlTypeName() == SqlTypeName.DATE) {
```

Comment:
Maybe also a comment here discussion why DATE is handled separately

Reply:
Same as above